### PR TITLE
[FIX] web_editor, website: correctly remove the image gallery snippet

### PR DESCRIPTION
**Pull request by loco-odoo**
Steps to reproduce the bug:
- Add an Image Gallery (IG) snippet on the page.
- Add 2 new images in the IG.
- Click on the first image of the IG to load its data.
- Click on the trash button to remove the snippet.
- Bug => The snippet is not removed (an image is removed instead).

When a snippet is removed, the `removeSnippet` function is called. The
problem is that the `call_for_each_child_snippet` will never resolve.
Two mechanisms are of interest to understand why: The first one is the
`updateCurrentSnippetEditorOverlay` function. Its goal is to destroy a
snippet each time its target is not in the DOM anymore. The second
mechanism is specific to the IG snippet: when an image of this snippet
is destroyed, the `slideshow` function goes through the remaining
images to update parameters. To do it, the function uses the
`_replaceContent` that empty the content of the carousel and then fill
it with new data.

When a snippet is removed, a `SnippetEditor` is created for each
element of it. In the case of the IG, a `SnippetEditor` is created for
each image of the the snippet. Because the first image already has a
`SnippetEditor` (because it has been clicked), the callback of
`call_for_each_child_snippet` is called to remove this image from the
IG snippet. The second mechanism explained before will then be called.
Meanwhile, a `SnippetEditor` will be created for the second image.
However, because the `_replaceContent` function emptied the content of
the carousel, the `updateCurrentSnippetEditorOverlay` function will
destroy the `SnippetEditor` of the second image as its target is not
considered present in the DOM anymore. Unfortunately, the
`call_for_each_child_snippet` still needed this `SnippetEditor` and
will never entirely resolve.

To solve this problem, the `removeSnippet` function is executed inside
a mutex. Because the mutex is also used by
`updateCurrentSnippetEditorOverlay`, we are sure that the snippetEditor
will not be destroyed while the `removeSnippet` is still running.

task-3147271

### CODE REVIEW

**Comment by qsm-odoo** [comment]
@robodoo delegate=detrouxdev

File: addons/website/static/tests/tours/snippet_image_gallery.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    trigger: ".o_select_media_dialog img[title='s_default_image2.jpg']",
+}, {
+    content: "Click on Add",
+    trigger: "button:has(span:contains('Add'))",
+}, {
+    content: "Click on the image of the Image Gallery snippet",
+    trigger: ".carousel img",
+}, {
+    content: "Click on Remove Block",
+    trigger: ".o_we_customize_panel we-button[title='Remove Block']",
```
Comment (detrouxdev) [inline addons/website/static/tests/tours/snippet_image_gallery.js:32]: This step should probably make sure that it targets the Image Gallery's Remove block button. If I'm not wrong, this step could also apply to a single image that we selected right? 🤔

**Review by detrouxdev — CHANGES_REQUESTED** [review]
Good commit message! 👍🏻 A few comments related to the tour! Don't forget to have a quick look at the utils provided by the website tour utils. Some of them are quite useful!

**Comment by loco-odoo** [comment]
@detrouxdev , thanks a lot for your review :pray: I adapted the code accordingly :+1:

**Comment by detrouxdev** [comment]
@loco-odoo The test that you introduce does not crash on my computer when I'm launching it with `--test-tags=.test_03_snippet_image_gallery` though it does crash when I'm running it directly with Chrome (doing `odoo.startTour`). Do you have the same issue? I assume it's linked to the fact that this is a race condition. Could you confirm that the test does fail with the fix?

Could you also pass the PR as Ready for review? After that I'll merge it 👍🏻 

Note: during my testing, I've realized that this also fix the issue of deleting a Dynamic Snippet while it's rendering. To test that, drop a dynamic snippet and preview a lot of options by hovering the mouse above them, when the loading appears, click on the remove button. It should lock the editor up.

**Comment by loco-odoo** [comment]
> The test that you introduce does not crash on my computer when I'm launching it with `--test-tags=.test_03_snippet_image_gallery` without your fix. Though it does crash when I'm running it directly with Chrome (doing `odoo.startTour`). Do you have the same issue? I assume it's linked to the fact that this is a race condition. Could you confirm that the test does fail with the fix?
@detrouxdev I confirm, on my laptop, the test fails when I am launching it without the fix and does not fail when I am launching it it with the fix :+1:

**Review by detrouxdev — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**